### ISSUE TITLE
Selective shouldShowNotificationOnPush

### ISSUE THREAD

**Issue by cvb941** [GetStream/stream-chat-android#3848]
Currently the shouldShowNotificationOnPush has no parameters and can only be used to disable the notifications globally.

It would be great if there were a parameter provided with the contents of the push notification, so that we could selectively decide whether to show or not show the notification, based on its parameters.

For example, we would like to disable showing chat notifications coming from a certain userId.

**Comment by adasiewiczr** [comment]
hey @cvb941,
thank you for reporting this.
`shouldShowNotificationOnPush` was intended to be used to disable notifications globally (for example when the app is in the foreground).
If you don't want to show notifications for a particular user it would be better to provide a custom `NotificationHandler` implementation and override `showNotification` or `onPushMessage` method.
Moreover, we can only provide `channelType`, `channelId` and `messageId` in the `shouldShowNotificationOnPush` function but `userId` is not present in the push notification message and we are fetching it from the server. 
To sum up, we would rather keep the `shouldShowNotificationOnPush` implementation as-is and recommend providing a custom `NotificationHandler` for this specific use case.
Let me know if this makes sense or if you have any further questions.
